source/ed/build_helper.cpp

```diff
@@ -397,6 +397,27 @@ void builder::connection(const std::string & name1, const std::string & name2, f
     connexion->departure->stop_point_connection_list.push_back(connexion);
     connexion->destination->stop_point_connection_list.push_back(connexion);
 }
+ double builder::get_co2_emission(const std::string& uri){
+     if (uri == "0x0"){
+         return 4.;
+     }
+     if (uri == "0x1"){
+         return 3.;
+     }
+     if (uri == "Bss"){
+         return 0.;
+     }
+     if (uri == "Bike"){
+         return 0.;
+     }
+     if (uri == "Bus"){
+         return 132.;
+     }
+     if (uri == "Car"){
+         return 184.;
+     }
+    return -1.;
+ }
```

Comment:
You forgot this one: please make this method a static function (`static double get_co2_emission(const std::strings& uri){`, then...